api/src/pcapi/core/users/factories.py

```diff
@@ -65,6 +65,8 @@ def _create(
         kwargs["password"] = crypto.hash_password(password)
         instance = super()._create(model_class, *args, **kwargs)
         instance.clearTextPassword = password
+        cls.on_init(instance)
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏ça me fait toujours peur des fonctions qui ressemblent à un `__init__`
Pourquoi ce choix ?

Reply:
Le fait est que j'ai eu besoin de manipuler les attributs de l'utilisateur, donc après la création de l'instance.
Au début je voulais le faire juste pour mettre une adresse email en fonction de l'`id` et après je me suis dit que ce comportement serait aussi utiles pour d'autres attributs.

Une fonction `@post_generation`est destinée aux attributs "externes" (qui ne sont pas des colonnes) il me semble, comme les fraud checks et le deposit.

Donc je me suis dit que vu la structure des classes des factories déjà implémentées ce serait pratique d'avoir une fonction qui soit adaptable en fonction du niveau du parcours d'inscription, et qui serait appelée directement à la création.
Ça évite aussi de répéter les instructions dans `_create`et `_build`, ou même d'_override_ ces deux dernières.

Si c'est juste le nom qui dérange, je vois aucun inconvénient à en mettre un plus explicite ;)

Reply:
Je pense que c'est surtout le nom qui me gène oui, init c'est "réservé" 

Il y a surement moyen de mutualiser une partie de `_create` et `_build` aussi si on veut éviter de se répéter